munemo/munemo.go

```diff
@@ -44,19 +54,63 @@ func newMunemo() *munemo {
 			"ya", "yi", "yu", "ye", "yo",
 			"za", "zi", "zu", "ze", "zo",
 		},
+		sign:           1,
+		symbol_values:  make(map[string]int),
 		negativeSymbol: "xa",
 		buffer:         new(bytes.Buffer),
 	}
+	for k, v := range m.symbols {
+		m.symbol_values[v] = k
+	}
+
+	return m
 }
 
 func (m *munemo) string() string {
 	return m.buffer.String()
 }
 
+func (m *munemo) int() int {
+	return m.number
+}
+
+func (m *munemo) decode(s string) {
+	arr := []byte(s)
+	m.buffer.Write(arr)
```

Comment:
this buffer is unused during the conversion. no need to write to it.

Reply:
Yeah. this code is a weird mix of functional and OO. I left that there as a start towards making it more OO'ish. But lemme rip it out. Maybe we should go more functional